sympy/simplify/tests/test_simplify.py

```diff
@@ -829,11 +829,11 @@ def test_issue_17141():
     assert simplify(acos(-I)**2*acos(I)**2) == \
            log(1 + sqrt(2))**4 + pi**2*log(1 + sqrt(2))**2/2 + pi**4/16
     assert simplify(2**acos(I)**2) == 2**((pi - 2*I*log(1 + sqrt(2)))**2/4)
+    p = 2**acos(I+1)**2
+    assert simplify(p) == p
 
     # However, for a complex number it still happens
     if PY3:
-        raises(RecursionError, lambda: simplify(2**acos(I+1)**2))
         raises(RecursionError, lambda: simplify((2**acos(I+1)**2).rewrite('log')))
```

Comment:
I should really have added these in a separate test with XFAIL. "Fortunately" it wasn't the only error in the test run.

Reply:
I wish that both were solved...the `as_real_imag` needs work (see #6679).

Reply:
If I could figure out (or someone else tell me, more likely) what `atan2(0, complex)` (see #17155) is, it should (probably) be solved. This is where it starts looping.

Reply:
If a and b are real then `1/(a +I*b)` cand be rewritten over a real value...why not just rewrite like that?

Reply:
I think that is what causes the recursion error.

Reply:
Well, not the original expression of course.... The problem is just to figure out when it works or not. (I cannot remember the details in this case but I wrote them down in some issue...)